src/extensions/crypt/rsa/rsa.c

```diff
@@ -193,21 +189,26 @@ void RSA_free(RSA_CTX *rsa_ctx)
 /**
  * @brief Use PKCS1.5 for decryption/verification.
  * @param ctx [in] The context
- * @param in_data [in] The data to encrypt (must be < modulus size-11)
- * @param out_data [out] The encrypted data.
+ * @param in_data [in] The data to decrypt (must be < modulus size-11)
+ * @param out_data [out] The decrypted data.
+ * @param out_len [int] The size of the decrypted buffer in bytes
  * @param is_decryption [in] Decryption or verify operation.
  * @return  The number of bytes that were originally encrypted. -1 on error.
  * @see http://www.rsasecurity.com/rsalabs/node.asp?id=2125
  */
-int RSA_decrypt(const RSA_CTX *ctx, const uint8_t *in_data,
-                            uint8_t *out_data, int is_decryption, int padding)
+int RSA_decrypt(const RSA_CTX *ctx, const uint8_t *in_data, 
+                            uint8_t *out_data, int out_len, int is_decryption)
```

Comment:
When they dropped the padding from the `RSA_encrypt`, you noticed and could drop the parameter.  When they dropped the padding from `RSA_decrypt`, they made a completely different integer parameter in the same spot.

When I think about the idea of these kinds of codebases and "security" it reminds me of The Princess Bride.  *"You keep using that word.  I don't think it means what you think it means."*  :-/

Reply:
Yes, that is terrible.

Reply:
Mathematician <> Programmer